### ISSUE TITLE
In Intellij IDEA test runner, the hierarchy is messed up when using IsolationMode.InstancePerLeaf

### ISSUE THREAD

**Issue by gdurandrexel** [kotest/kotest#1555]
**Which version of Kotest are you using** : 4.1.1
Using kotest plugin 1.0.5

In a FreeSpec with nested tests and IsolationMode.InstancePerLeaf, the hierarchy of tests in the test runner does not reflect the actual hierarchy: each leaf test run is nested inside the previous one. Works fine with InstancePerTest.
```
composite - {
  test1 { ... }
  test2 { ... }
  test3 { ... }
}
```
is displayed as:
```
composite
  test1
    composite
      test2
        composite
          test3
}
```

**Comment by sksamuel** [comment]
Duplicate of #1547 fixed with #1554

**Comment by sksamuel** [comment]
Fixed in 4.1.2
